### ISSUE TITLE
Failure to interpolate images between two images smoothly

### ISSUE THREAD

**Issue by MHX1203** [rosinality/stylegan2-pytorch#312]
I have alreay project two image to the latent space, and the projected result is well and real.

but when i use the latent vector to interpolate new vector, and use new vector to generate images between two original images. I got bad result, the generated images is not real, and contain some artifact. 

i 'm  wondering if i use the wrong interplotion ways.

`import argparse
import math
import os
from cv2 import resize
import torch
from torch import optim
from torch.nn import functional as F
from torchvision import transforms
from PIL import Image
from tqdm import tqdm
import shutil
import lpips
from model import Generator
def noise_regularize(noises):
    loss = 0
    for noise in noises:
        size = noise.shape[2]
        while True:
            loss = (
                loss
                + (noise * torch.roll(noise, shifts=1, dims=3)).mean().pow(2)
                + (noise * torch.roll(noise, shifts=1, dims=2)).mean().pow(2)
            )
            if size <= 8:
                break
            noise = noise.reshape([-1, 1, size // 2, 2, size // 2, 2])
            noise = noise.mean([3, 5])
            size //= 2
    return loss
def noise_normalize_(noises):
    for noise in noises:
        mean = noise.mean()
        std = noise.std()
        noise.data.add_(-mean).div_(std)
def get_lr(t, initial_lr, rampdown=0.25, rampup=0.05):
    lr_ramp = min(1, (1 - t) / rampdown)
    lr_ramp = 0.5 - 0.5 * math.cos(lr_ramp * math.pi)
    lr_ramp = lr_ramp * min(1, t / rampup)
    return initial_lr * lr_ramp

def latent_noise(latent, strength):
    noise = torch.randn_like(latent) * strength
    return latent + noise
def make_image(tensor):
    return (
        tensor.detach()
        .clamp_(min=-1, max=1)
        .add(1)
        .div_(2)
        .mul(255)
        .type(torch.uint8)
        .permute(0, 2, 3, 1)
        .to("cpu")
        .numpy()
    )
if __name__ == "__main__":
    device = "cuda"
    parser = argparse.ArgumentParser(
        description="Image projector to the generator latent spaces"
    )
    parser.add_argument(
        "--ckpt", type=str, default="checkpoint/070000.pt", help="path to the model checkpoint"
    )
    parser.add_argument(
        "--num_inter",
        type=int,
        help="interpolated image number",
        default=10
    )

    args = parser.parse_args()

    n_mean_latent = 10000
    resize = args.size

    g_ema = Generator(args.size, 512, 8)
    g_ema.load_state_dict(torch.load(args.ckpt)["g_ema"], strict=False)
    g_ema.eval()
    g_ema = g_ema.to(device)
    # saved latent vectors and noises
    pt_value="/mnt/d/dy/stylegan2-pytorch/second_job/test_oursandstone.pt"

    result_file=torch.load(pt_value)
    keys = list(result_file.keys())

    for index in range(len(keys) - 1):
        # first image
        temp_start_file=result_file[keys[index]]
        temp_start_noise=temp_start_file['noise']
        temp_start_latent=temp_start_file['latent']
        temp_start_latent=torch.unsqueeze(temp_start_latent,0)

        # second image
        temp_end_file=result_file[keys[index + 1]]
        temp_end_noise=temp_end_file['noise']
        temp_end_latent=temp_end_file['latent']
        temp_end_latent=torch.unsqueeze(temp_end_latent,0)

        outdir = f"/mnt/d/dy/stylegan2-pytorch/second_job/test_oursandstone_output/{os.path.split(keys[index])[-1][:-4]}-{os.path.split(keys[index + 1])[-1][:-4]}_dy"
        print("Saved to: ", outdir)
        os.makedirs(outdir, exist_ok=True)

        for idx, i in tqdm(enumerate(torch.linspace(0, 1, args.num_inter)), desc=f"from image {keys[index]} ---> {keys[index + 1]}"):
            inter_latent = temp_start_latent * (1 - i) + temp_end_latent * i
            inter_noise = [temp_start_noise[_] * (1 - i) + temp_end_noise[_] * i for _ in range(len(temp_start_noise))]

            img_gen, _ = g_ema([inter_latent], input_is_latent=True, noise=inter_noise)

            img_name = os.path.join(outdir,f'output_{idx}.png')
            img_ar = make_image(img_gen)
            pil_img = Image.fromarray(img_ar[0])
            pil_img.save(img_name)

        shutil.copy(keys[index], f"{outdir}/0.png")
        shutil.copy(keys[index + 1], f"{outdir}/{args.num_inter}.png")
`
Anyone can tell me what is the correct way to interplotate images between two arbitrary images.  

thanks.

**Comment by rosinality** [comment]
Is it okay on i = 0 or 1? If it works then it is hard to say why it does not works.

**Comment by MHX1203** [comment]
yes, result is ok when i =0  or i = 1. just the middle result is blur, and have low quality.

**Comment by MHX1203** [comment]
yes, result is ok when i =0  or i = 1. just the middle result is blur, and have low quality.
